src/libraries/math/SafeMathUint256.sol

```diff
@@ -45,4 +44,22 @@ library SafeMathUint256 {
             return b;
         }
     }
+
+    function safeToMultiply(uint256 a, uint256 b) internal constant returns (bool) {
+        uint256 c = a * b;
+        if (a == 0 || c / a == b) {
+            return true;
+        }
+        return false;
+    }
+
+    // Float [fixed point] Operations
+    function fxpMul(uint256 a, uint256 b) internal constant returns (uint256) {
+        require(safeToMultiply(a, b));
```

Comment:
With the `safeToMultiply` check extracted out, this ends up doing the arithmetic twice.  Consider merging in that method so we don't have to dot he math twice.  Though, if you have a compelling reason for keeping it extracted the argument for gas savings is pretty insignificant.

Reply:
Consider just changing this to `div(mul(a, b), 10**18)`

Reply:
The main reason I did it was because the logic was shared between mul() and fxpMul(), but I think the savings in gas is a much more compelling reason to do the logic inline.